src/shared/cloudformation/templateRegistryManager.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { CloudFormationTemplateRegistry } from './templateRegistry'
+
+export class CloudFormationTemplateRegistryManager implements vscode.Disposable {
+
+    // we may want to move this to non-static to handle globs from any added watchers
+    // consider creating watchers from this class?
```

Comment:
This seems appropriate. Rebuilding the registry needs to re-use the globs, so it makes sense to push the globs of interest up to the caller.

In that case...

* TEMPLATE_FILE_GLOB_PATTERN  moves to activation.ts
* addWatcher becomes private
* new method: addTemplateGlob(vscode.globPattern)
  * maintains list of globs
  * creates watcher, retains watcher in this.disposables, calls addWatcher
* rebuildRegistry uses list of globs

This class would either be tested in integration tests, or by mocking parts of the VS Code api (create file watcher), which is reasonable, because this class purpose is to deal with workspace related logic.